File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        if(head==NULL || (head->next==NULL && n>=1))
        return NULL;
        ListNode *slow = new ListNode(0);
        slow->next = head;
        ListNode * fast = head;
        while(n>0)
        {
            fast = fast->next;
            n--;
        }
        while(fast!=NULL)
        {
            slow = slow->next;
            fast = fast->next;
        }
        ListNode *todelete = slow->next;
        if(todelete == head)
        {
            head = head->next;
        }
        ListNode *item = slow->next->next;
        slow->next = item;
        return head;


    }
};

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        if head is None:
            return None
        dummy = ListNode(0,head)
        slow = dummy 
        fast = head
        while n>0:
            fast = fast.next
            n=n-1
        while fast:
            fast = fast.next
            slow = slow.next
        slow.next = slow.next.next
        return dummy.next
        